salalib/pointdata.cpp

```diff
@@ -517,12 +517,15 @@ bool PointMap::blockLines()
          PixelRef curs = PixelRef( i, j );
          Point& pt = getPoint( curs );
          QtRegion viewport = regionate( curs, 1e-10 );
-         for (size_t k = pt.m_lines.size() - 1; k != paftl::npos; k--) {
-            if (!pt.m_lines.value(k).crop( viewport )) {
-               // the pixelation is fairly rough to make sure that no point is missed: this just
-               // clears up if any point has been added in error:
-               pt.m_lines.remove_at(k);
-            }
+         std::map<int, Line>::iterator iter = pt.m_lines.begin(), end = pt.m_lines.end();
+         for(; iter != end; ) {

```

Comment:
Should probably evaluate `.end()` in the for statement - depending on the stl implementation, erase might change where end is pointing to.